### PR TITLE
Add enforcePodSecurityPolicy to DC spec

### LINKED ISSUE #5225: Allow enforcing PSP on datacenter level
**User Story**
Analogous to #4298, but for PSPs.

**Acceptance criteria**
- [ ] A new per-datacenter setting `enforcePSP` will cause all clusters created within the given datacenter to have PSP enabled.
- [ ] This will be added to both old file-based DCs, as well as DC CRDs.
- [ ] The nornal per-cluster setting is ignored.
- [ ] The value of `enforceAuditLogging` is sent over the API then datacenter(s) are fetched.

### PR DESCRIPTION
**Pull request by moelsayed**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5225

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
It's now possible to enable PodSecurityPolicy on a datacenter level. 
```

### CODE REVIEW

File: api/pkg/handler/v1/admin/seed_test.go [hunk 1]
```diff
@@ -174,7 +174,7 @@ func TestUpdateSeedEndpoint(t *testing.T) {
...
+			expectedResponse:       `{"name":"us-central1","spec":{"country":"US","location":"us-central","kubeconfig":{},"datacenters":{"audited-dc":{"country":"Germany","location":"Finanzamt Castle","node":{},"spec":{"fake":{},"enforceAuditLogging":true,"enforcePodSecurityPolicy":false}},"fake-dc":{"country":"Germany","location":"Henriks basement","node":{},"spec":{"fake":{},"enforceAuditLogging":false,"enforcePodSecurityPolicy":false}},"private-do1":{"country":"NL","location":"US ","node":{},"spec":{"digitalocean":{"region":"ams2"},"enforceAuditLogging":false,"enforcePodSecurityPolicy":false}},"regular-do1":{"country":"NL","location":"Amsterdam","node":{},"spec":{"digitalocean":{"region":"ams2"},"enforceAuditLogging":false,"enforcePodSecurityPolicy":false}},"restricted-fake-dc":{"country":"NL","location":"Amsterdam","node":{},"spec":{"fake":{},"requiredEmailDomain":"example.com","enforceAuditLogging":false,"enforcePodSecurityPolicy":false}},"restricted-fake-dc2":{"country":"NL","location":"Amsterdam","node":{},"spec":{"fake":{},"requiredEmailDomains":["example.com","noifsdsd.org"],"enforceAuditLogging":true,"enforcePodSecurityPolicy":false}}}}}`,
```
Comment (moadqassem) [inline api/pkg/handler/v1/admin/seed_test.go:177]: I know that what is in the body will be decoded to what you specify beforehand, but maybe it makes sense to try setting the `enforcePodSecurityPolicy` to `true` once :)

File: api/pkg/handler/v1/dc/datacenter_test.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ func TestDatacentersEndpoint(t *testing.T) {
...
+	test.CompareWithResult(t, res, ` [{"metadata":{"name":"audited-dc","resourceVersion":"1"},"spec":{"seed":"us-central1","country":"Germany","location":"Finanzamt Castle","provider":"fake","enforceAuditLogging":true,"enforcePodSecurityPolicy":false}},{"metadata":{"name":"fake-dc","resourceVersion":"1"},"spec":{"seed":"us-central1","country":"Germany","location":"Henriks basement","provider":"fake","enforceAuditLogging":false,"enforcePodSecurityPolicy":false}},{"metadata":{"name":"private-do1","resourceVersion":"1"},"spec":{"seed":"us-central1","country":"NL","location":"US ","provider":"digitalocean","digitalocean":{"region":"ams2"},"enforceAuditLogging":false,"enforcePodSecurityPolicy":false}},{"metadata":{"name":"regular-do1","resourceVersion":"1"},"spec":{"seed":"us-central1","country":"NL","location":"Amsterdam","provider":"digitalocean","digitalocean":{"region":"ams2"},"enforceAuditLogging":false,"enforcePodSecurityPolicy":false}},{"metadata":{"name":"us-central1","resourceVersion":"1"},"spec":{"seed":"","enforceAuditLogging":false,"enforcePodSecurityPolicy":false},"seed":true}]
```
Comment (moadqassem) [inline api/pkg/handler/v1/dc/datacenter_test.go]: Like the previous comment, same goes here, maybe enabling the `enforcePodSecurityPolicy` for one datacenter spec.

File: api/pkg/handler/v1/admin/seed_test.go (hunk 1 above)
Comment (moelsayed) [reply api/pkg/handler/v1/admin/seed_test.go:177]: Done

File: api/pkg/handler/v1/dc/datacenter_test.go (hunk 2 above)
Comment (moelsayed) [reply api/pkg/handler/v1/dc/datacenter_test.go]: done

**Pull request merged**